File: Difficulty: Easy/Sort 0s, 1s and 2s/sort-0s-1s-and-2s.cpp
//{ Driver Code Starts
#include <bits/stdc++.h>
using namespace std;


// } Driver Code Ends
class Solution {
  public:
    void sort012(vector<int>& arr) {
        // code here
        int cz=0,co=0;
        int n=arr.size();
        for(int i=0;i<n;i++)
        {
            if(arr[i]==0) cz++;
            else if(arr[i]==1) co++;
        }
        for(int i=0;i<cz;i++) arr[i]=0;
        for(int i=cz;i<co+cz;i++) arr[i]=1;
        for(int i=co+cz;i<n;i++) arr[i]=2;
    }
};


//{ Driver Code Starts.
int main() {

    int t;
    cin >> t;
    cin.ignore();
    while (t--) {

        vector<int> a;
        string input;
        int num;

        getline(cin, input);
        stringstream s2(input);
        while (s2 >> num) {
            a.push_back(num);
        }
        Solution ob;
        ob.sort012(a);

        int n = a.size();
        for (int i = 0; i < n; i++) {
            cout << a[i] << " ";
        }

        cout << endl;
        cout << "~" << endl;
    }
    return 0;
}

// } Driver Code Ends

File: Difficulty: Medium/Sort 0s, 1s and 2s/sort-0s-1s-and-2s.py
class Solution:
    def sort012(self, arr):
        # code here
        cntZero, cntOne = 0,0
        for i in range(0,len(arr)):
            if arr[i]==0 :
                cntZero+=1
            elif arr[i]==1 :
                cntOne+=1
                
        for i in range(0,len(arr)):
            if i<cntZero :
                arr[i]=0
            elif i<cntZero+cntOne :
                arr[i]=1
            else :
                arr[i]=2
        